src/btree/bt_compact.c

```diff
@@ -24,13 +25,15 @@ __compact_rewrite(WT_SESSION_IMPL *session, WT_REF *ref, bool *skipp)
 
     bm = S2BT(session)->bm;
 
+    /* If the page is clean, test the original addresses. */
+    if (__wt_page_evict_clean(ref->page))
+        return (__wt_ref_addr_copy(session, ref, &addr) ?
+            bm->compact_page_skip(bm, session, addr.addr, addr.size, skipp) :
+            0);
```

Comment:
Does this mean that if we're unable to copy the `addr`, then we don't do anything but signal success?